apps/web/src/utils/badge.tsx

```diff
@@ -0,0 +1,69 @@
+import Tooltip from "@web/components/Tooltip";
+import { Permer } from "permer";
+import { Code, Plus, Tool, Zap } from "react-feather";
+import { SelfUser } from "../../../../packages/commons/types";
+import { styled } from "@web/stitches.config";
+
+const badges = [
+  {
+    id: "admin",
+    tooltip: "imperialb.in Staff",
+    icon: <Tool color="dark red" fill="red" />,
+  },
+
+  {
+    id: "beta-tester",
+    tooltip: "Beta Tester",
+    icon: <Code color="dark purple" fill="purple" />,
+  },
+
+  {
+    id: "early-adopter",
+    tooltip: "Early Adopter",
+    icon: <Zap color="dark orange" fill="orange" />,
+  },
+
+  {
+    id: "member-plus",
+    tooltip: "Supporter of imperialb.in",
+    icon: <Plus color="pink" fill="pink" />,
+  },
+] as const;
+
+const badgeChecker = new Permer(
```

Comment:
this actually wont work in this case, mainly because the array of badges is out of order with the actual permer in `@imperial/commons` which is important because they're bitwise operations.

is there a way we can use the permer from commons?

Reply:
The main idea behind implementing it like this was to make it so we could implement more badges extremely easily and not bound the implementation to just roles unless you don't want that.

If you just want to display roles then yes we can switch to permer from commons.